contracts/Uniswap/UniswapAnchoredView.sol

```diff
@@ -79,11 +79,16 @@ contract UniswapAnchoredView is UniswapConfig {
         for (uint i = 0; i < configs.length; i++) {
             TokenConfig memory config = configs[i];
             address uniswapMarket = config.uniswapMarket;
-            uint cumulativePrice = currentCumulativePrice(config);
-            oldObservations[uniswapMarket].timestamp = block.timestamp;
-            newObservations[uniswapMarket].timestamp = block.timestamp;
-            oldObservations[uniswapMarket].acc = cumulativePrice;
-            newObservations[uniswapMarket].acc = cumulativePrice;
+            if (uniswapMarket != address(0)) {
+                require(config.priceSource == PriceSource.REPORTER, "only reported prices utilize an anchor");
+                uint cumulativePrice = currentCumulativePrice(config);
+                oldObservations[uniswapMarket].timestamp = block.timestamp;
+                newObservations[uniswapMarket].timestamp = block.timestamp;
+                oldObservations[uniswapMarket].acc = cumulativePrice;
+                newObservations[uniswapMarket].acc = cumulativePrice;
+            } else {
+                require(config.symbolHash != ethHash, "ETH must have a uniswap market specified");
```

Comment:
what about FIXED_USD markets? which have no uniswapMarket but arent ETH?

Reply:
Those don't need a uniswap market right? They would match not eth and would not revert